src/x3f_image.c

```diff
@@ -166,16 +168,56 @@
   return 1;
 }
 
+/* extern */ int x3f_get_camf_rect(x3f_t *x3f, char *name,
+				   x3f_area16_t *image, int rescale,
+				   uint32_t *rect)
+{
+  if (!x3f_get_camf_matrix(x3f, name, 4, 0, 0, M_UINT, rect))
+    return 0;
+
+  return x3f_transform_rect_to_keep_image(x3f, image, rescale, rect);
+}
+
+/* extern */ int x3f_crop_area_column(x3f_t *x3f, char *name,
+				      x3f_area16_t *image, int rescale,
+				      x3f_area16_t *crop)
+{
+  uint32_t rect[4];
+  uint32_t column[4];
+
+  if (!x3f_get_camf_matrix(x3f, "DarkShieldColRange", 2, 2, 0, M_UINT, column))
+    return 0;
+
+  rect[1] = 0;           /* Cropped automatically later */
+  rect[3] = UINT32_MAX;  /* Cropped automatically later */
+
+  if (!strcmp(name, "FakeDarkShieldLeft")) {
```

Comment:
I agree on the fake thing. Ill remove it. But ... the function is so special, so it do not need the DarkShieldColRanges parameter. It can be implicit.

Then, I am a bit spoliled by languages with atoms, and I am 100% fed up with inventing defines for left and right, and I really do not fancy sending 0 and 1. The strings "LEFT" and "RIGHT" would just feel nice :) OK - then you need to use strcmp. je je ... ill add two global defines for COL_LEFT_RECT and COL_RIGHT_RECT or some such. But ... I do not like it. Not at all.